cloudformation.yaml

```diff
@@ -29,14 +29,44 @@ Parameters:
     Default: 7
   SsmOutputBucketName:
     Type: String
-    Description: Bucket used to store SSM command output. This bucket must already exist and the instances which receive SSM commands must have PutObject permissions for this bucket.
+    Description: OPTIONAL! Bucket used to store SSM command output. The instances which receive SSM commands must have PutObject permissions for this bucket. If this is not provided a bucket will be created as part of this stack.
+    Default: ""
   SNSTopicForAlerts:
     Type: String
     Description: The name of the SNS topic used for alerting in the case of a failed rotation attempt.
 
 
+Conditions:
+  ShouldCreateSsmOutputBucket: !Equals
+    - !Ref SsmOutputBucketName
+    - ""
+
 Resources:
 
+  NodeRotationSsmOutputBucket:
+    Type: AWS::S3::Bucket
+    Condition: ShouldCreateSsmOutputBucket
+    Properties:
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+          - ServerSideEncryptionByDefault:
+              SSEAlgorithm: aws:kms
+      PublicAccessBlockConfiguration:
+        BlockPublicAcls: true
+        IgnorePublicAcls: true
+        BlockPublicPolicy: true
+        RestrictPublicBuckets: true
+      Tags:
+        - { Key: Stack, Value: !Ref Stack }
+        - { Key: Stage, Value: !Ref Stage }
+
+  NodeRotationBucketSsmParameter:
+    Type: AWS::SSM::Parameter
+    Properties:
+      Name: node-rotation-ssm-output-bucket-name
```

Comment:
As mentioned elsewhere, this should be part of a hierarchy rather than in the root. Under `/account/services` might do for now although this isn't strictly an account service in my mind. Perhaps we need a better pattern for general service registration/discovery.